File: 01/b.hs
{-# LANGUAGE OverloadedStrings #-}

import Data.Text (pack, splitOn, unpack)
import Data.List (sort)

main :: IO ()
main = do
    text <- readFile "input.txt"
    let chunks = splitOn "\n\n" $ pack text
    let parse = map (read . unpack) . filter (/= "") . splitOn "\n"
    let intChunks = map parse chunks :: [[Int]]
    let greatest = sum $ take 3 $ reverse $ sort $ map sum intChunks
    print greatest

File: 01/b.swift
import Foundation

let text = try! String(contentsOfFile: "input.txt")
let chunks = text.components(separatedBy: "\n\n")

let intChunks = chunks.map { chunk in
    chunk
        .components(separatedBy: .newlines)
        .filter { !$0.isEmpty }
        .map { Int($0)! }
}

let topThree = intChunks.map({ $0.reduce(0, +) }).sorted(by: >).prefix(3)
let sum = topThree.reduce(0, +)
print(sum)
